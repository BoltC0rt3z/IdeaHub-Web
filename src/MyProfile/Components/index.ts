import ModalHeader from './ModalHeader';
import ProfileForm from './ProfileForm';

export { ModalHeader, ProfileForm };
