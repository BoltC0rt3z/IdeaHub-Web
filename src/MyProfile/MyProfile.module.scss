@import '../styles/styles.module.scss';

.myProfile {
  background: $color-light;

  .coverImageContainer {
    height: 200px;

    .coverImage {
      width: 100%;
      height: 100%;
      object-fit: cover;
    }
  }

  .profileImgContainer {
    padding-left: 16px;
    padding-right: 16px;

    .imageContainer {
      margin-top: -120px;

      .image {
        width: 136px;
        height: 136px;
        object-fit: cover;
        border-radius: 50%;
        border: 5px solid $color-light;
      }

      @include respond-to('xs-l') {
        margin-top: -80px;

        .image {
          width: 80px;
          height: 80px;
        }
      }
    }

    .editProfileButton {
      margin-top: -60px;
    }
  }
}

.nameContainer {
  color: $color-dark-blue-lighter;
  padding-left: 28px;
  padding-bottom: 12px;

  .name {
    font-weight: bold;
    font-size: 20px;
  }
}

.ideasContainer {
  padding-top: 20px;
}
