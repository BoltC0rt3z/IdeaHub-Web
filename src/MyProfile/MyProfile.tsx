import React from 'react';
import { Row, Col } from 'antd';

import { coverImage, profileImage } from '../MockData';
import styles from './MyProfile.module.scss';
import { Button, Modal } from '../Components';
import { Ideas } from '../Ideas';
import { ModalHeader, ProfileForm } from './Components';

const MyProfile: React.FC = () => {
  return (
    <Row>
      <Modal title={<ModalHeader />} centered={true} closeIcon={null} className={styles.modal} visible footer={null}>
        <ProfileForm />
      </Modal>
      <Row>
        <Col lg={6} md={4} sm={0} />
        <Col lg={12} md={16} sm={24} className={styles.myProfile}>
          <Row className={styles.coverImageContainer}>
            <img src={coverImage} alt="Cover" className={styles.coverImage} />
          </Row>
          <Row type="flex" justify="space-between" className={styles.profileImgContainer}>
            <Col>
              <div className={styles.imageContainer}>
                <img src={profileImage} alt="Cover" className={styles.image} />
              </div>
            </Col>
            <Col className={styles.editProfileButton}>
              <Button size="large">Edit profile</Button>
            </Col>
          </Row>
          <Row className={styles.nameContainer}>
            <Row className={styles.name}>Brian Ogutu</Row>
            <Row>Software Engineer Tech Lead</Row>
          </Row>
        </Col>
        <Col lg={6} md={4} sm={0} />
      </Row>
      <Row className={styles.ideasContainer}>
        <Ideas hidePostIdea={true} />
      </Row>
    </Row>
  );
};

export default MyProfile;
