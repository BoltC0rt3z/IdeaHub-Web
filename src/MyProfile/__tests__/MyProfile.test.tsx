import React from 'react';
import { mount, ReactWrapper } from 'enzyme';
import MyProfile from '../MyProfile';

describe('Tests for my profile container', () => {
  const wrapper: ReactWrapper = mount(<MyProfile />);

  it('Should render my profile without crushing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
