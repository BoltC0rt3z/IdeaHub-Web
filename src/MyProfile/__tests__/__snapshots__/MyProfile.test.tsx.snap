// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for my profile container Should render my profile without crushing 1`] = `ReactWrapper {}`;
